packages/sdk/src/browser-wallets/xverse/__tests__/index.spec.ts

```diff
@@ -0,0 +1,150 @@
+// @vitest-environment happy-dom
+import type { GetAddressOptions, GetAddressResponse } from "sats-connect";
+import * as satsConnect from "sats-connect";
+
+import { WalletAddress } from "../../types";
+import { getAddresses, isInstalled } from "..";
+
+vi.mock("sats-connect", async (originalImport) => {
+  const mod = (await originalImport()) as typeof satsConnect;
+  return {
+    __esModule: true,
+    ...mod,
+  };
+});
+
+describe("Xverse Wallet", () => {
+  afterEach(() => {
+    vi.unstubAllGlobals();
+  });
+
+  describe("isInstalled", () => {
+    test("should return true if installed", () => {
+      vi.stubGlobal("BitcoinProvider", {});
+      expect(typeof window).not.toBeUndefined();
+      expect(isInstalled()).toBeTruthy();
+    });
+
+    test("should return false if not installed", () => {
+      expect(typeof window).not.toBeUndefined();
+      expect(isInstalled()).toBeFalsy();
+    });
+  });
+
+  describe("getAddresses", () => {
+    afterEach(() => {
+      vi.unstubAllGlobals();
+      vi.resetAllMocks();
+    });
+
+    test("should return address from mainnet", () => {
+      const mockData: WalletAddress[] = [
+        {
+          // https://bitcoin.design/guide/glossary/address/
+          address:
+            "bc1p5d7rjq7g6rdk2yhzks9smlaqtedr4dekq08ge8ztwac72sfr9rusxg3297",
+          publicKey:
+            "03b9907521ddb85e0e6a37622b7c685efbdc8ae53a334928adbd12cf204ad4e717",
+          format: "taproot",
+        },
+        {
+          // https://bitcoin.design/guide/glossary/address/
+          address: "bc1qar0srrr7xfkvy5l643lydnw9re59gtzzwf5mdq",
+          publicKey:
+            "0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798",
+          format: "segwit",
+        },
+      ];
+      const mockResponse: GetAddressResponse = {
+        addresses: [
+          {
+            address:
+              "bc1p5d7rjq7g6rdk2yhzks9smlaqtedr4dekq08ge8ztwac72sfr9rusxg3297",
+            publicKey:
+              "b9907521ddb85e0e6a37622b7c685efbdc8ae53a334928adbd12cf204ad4e717",
+            purpose: satsConnect.AddressPurpose.Ordinals,
+          },
+          {
+            address: "bc1qar0srrr7xfkvy5l643lydnw9re59gtzzwf5mdq",
+            publicKey:
+              "0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798",
+            purpose: satsConnect.AddressPurpose.Payment,
+          },
+        ],
+      };
+
+      vi.stubGlobal("BitcoinProvider", {});
+
+      const getAddressSpy = vi.spyOn(satsConnect, "getAddress");
+
+      getAddressSpy.mockImplementation((options: GetAddressOptions) => {
+        options.onFinish(mockResponse);
+        return Promise.resolve();
+      });
```

Comment:
this seems to be repeated maybe we can mock it on the top level once only

Reply:
It is repeated but not used for all, hence we can't just do one top level.